udata_hydra/analysis/resource.py

```diff
@@ -187,11 +187,18 @@ async def detect_resource_change_from_checksum(
         "analysis:last-modified-detection": "computed-checksum",
     }
     """
-    if last_check and last_check.get("checksum") != new_checksum:
-        return Change.HAS_CHANGED, {
-            "analysis:last-modified-at": datetime.now(timezone.utc).isoformat(),
-            "analysis:last-modified-detection": "computed-checksum",
-        }
+    if last_check:
```

Comment:
do we need to make sure last_check checksum is not None?
```suggestion
    if last_check and last_check.get("checksum"):
```

Reply:
Next line is `if last_check.get("checksum") != new_checksum:`, is it included?

Reply:
what if `last_check.get("checksum")` is `None` and `new_checksum` is `9985dbbf352fa2bb4b49b517ec95b4edfd1f1baf`, the comparison isn't fair?

Reply:
If we have a `new_checksum` then it means we have downloaded the file, but last time we didn't, so I'd say very little chance they're the same? But it's cleaner anyway, I'll add that